packages/opentelemetry-sdk-trace-base/src/config.ts

```diff
@@ -86,48 +89,31 @@ export function buildSamplerFromEnv(
         root: new AlwaysOffSampler(),
       });
     case TracesSamplerValues.TraceIdRatio:
-      return new TraceIdRatioBasedSampler(
-        getSamplerProbabilityFromEnv(environment)
-      );
+      return new TraceIdRatioBasedSampler(getSamplerProbabilityFromEnv());
     case TracesSamplerValues.ParentBasedTraceIdRatio:
       return new ParentBasedSampler({
-        root: new TraceIdRatioBasedSampler(
-          getSamplerProbabilityFromEnv(environment)
-        ),
+        root: new TraceIdRatioBasedSampler(getSamplerProbabilityFromEnv()),
       });
     default:
       diag.error(
-        `OTEL_TRACES_SAMPLER value "${environment.OTEL_TRACES_SAMPLER} invalid, defaulting to ${FALLBACK_OTEL_TRACES_SAMPLER}".`
+        `OTEL_TRACES_SAMPLER value "${sampler} invalid, defaulting to ${FALLBACK_OTEL_TRACES_SAMPLER}".`
       );
       return new AlwaysOnSampler();
```

Comment:
Unrelated to this PR: it strikes me as wrong that `ParentBasedAlwaysOn` is the default if not specified, but a different value (`AlwaysOnSampler`) is the fallback if the given value is unknown.

I opened https://github.com/open-telemetry/opentelemetry-js/issues/5476 for this.

Reply:
that is indeed wrong - thank you for opening the issue.